src/libs/actions/Report.js

```diff
@@ -1250,32 +1283,69 @@ function editReportComment(reportID, originalReportAction, textForNewComment) {
         return;
     }
 
-    // Optimistically update the report action with the new message
+    // Optimistically update the reportAction with the new message
     const sequenceNumber = originalReportAction.sequenceNumber;
-    const newReportAction = {...originalReportAction};
-    const actionToMerge = {};
-    newReportAction.message[0].isEdited = true;
-    newReportAction.message[0].html = htmlForNewComment;
-    newReportAction.message[0].text = parser.htmlToText(htmlForNewComment);
-    actionToMerge[sequenceNumber] = newReportAction;
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionToMerge);
-
-    // Persist the updated report comment
-    DeprecatedAPI.Report_EditComment({
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            message: [{
+                isEdited: true,
+                html: htmlForNewComment,
+                text: textForNewComment,
+            }],
+        },
+    };
+
+    // Optimistically update the report's last message if the user edited it
+    const highestSequenceNumber = getMaxSequenceNumber(reportID);
+    const optimisticReport = {};
+    if (sequenceNumber === highestSequenceNumber) {
```

Comment:
As we saw in the web side it's not enough to target the `maxSequenceNumber`. It needs to be the last visible I think.